BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -463,6 +464,14 @@ SecStartup2 (
     }
   }
 
+  // Copy device table to memory
+  if (LdrGlobal->DeviceTable != NULL) {
+    DeviceTable = (SBL_DEVICE_TABLE *) LdrGlobal->DeviceTable;
+    AllocateLen = DeviceTable->DeviceNumber * sizeof (SBL_DEVICE) + sizeof (SBL_DEVICE_TABLE);
+    LdrGlobal->DeviceTable = AllocatePool (AllocateLen);
```

Comment:
Do we need to allocate memory if DeviceNumber is 0?

Reply:
Device table should not be empty since boot option and others will depend on it.  And I don't see any issue to allocate memory if device number is really 0.